examples/morpheusvm/auth/bls.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package auth
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/crypto"
+	"github.com/ava-labs/hypersdk/crypto/bls"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/consts"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/storage"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+var _ chain.Auth = (*BLS)(nil)
+
+const (
+	BLSComputeUnits = 10
+	BLSSize         = bls.PublicKeyLen + bls.SignatureLen
+)
+
+type BLS struct {
+	Signer    bls.PublicKey `json:"signer"`
+	Signature bls.Signature `json:"signature"`
+
+	addr codec.Address
+}
+
+func (d *BLS) address() codec.Address {
+	if d.addr == codec.EmptyAddress {
+		d.addr = NewBLSAddress(d.Signer)
+	}
+	return d.addr
+}
+
+func (*BLS) GetTypeID() uint8 {
+	return consts.BLSID
+}
+
+func (*BLS) MaxComputeUnits(chain.Rules) uint64 {
+	return BLSComputeUnits
+}
+
+func (*BLS) ValidRange(chain.Rules) (int64, int64) {
+	return -1, -1
+}
+
+func (d *BLS) StateKeys() []string {
+	return []string{
+		string(storage.BalanceKey(d.address())),
+	}
+}
+
+func (d *BLS) AsyncVerify(msg []byte) error {
+	if !bls.Verify(msg, &d.Signer, &d.Signature) {
+		return crypto.ErrInvalidSignature
+	}
+	return nil
+}
+
+func (d *BLS) Verify(
+	_ context.Context,
+	r chain.Rules,
+	_ state.Immutable,
+	_ chain.Action,
+) (uint64, error) {
+	// We don't do anything during verify (there is no additional state to check
+	// to authorize the signer other than verifying the signature)
+	return d.MaxComputeUnits(r), nil
+}
+
+func (d *BLS) Actor() codec.Address {
+	return d.address()
+}
+
+func (d *BLS) Sponsor() codec.Address {
+	return d.address()
+}
+
+func (*BLS) Size() int {
+	return BLSSize
+}
+
+func (d *BLS) Marshal(p *codec.Packer) {
+	p.PackFixedBytes(bls.SerializePublicKey(&d.Signer))
+	p.PackFixedBytes(bls.SignatureToBytes(&d.Signature))
+}
+
+func UnmarshalBLS(p *codec.Packer, _ *warp.Message) (chain.Auth, error) {
+	var d BLS
+
+	signer := bls.SerializePublicKey(&d.Signer)
```

Comment:
You can just create a byte array of the size you are expecting and pass that into `Deserialize*` instead of doing this (which is going to be much less efficient)